afkak/test/test_consumer.py

```diff
@@ -784,6 +782,27 @@ def test_consumer_error_during_offset(self):
         consumer.stop()
         self.assertEqual(self.successResultOf(d), (None, None))
 
+    def test_consumer_offset_out_of_range_error(self):
+        topic = 'offset_out_of_range_error'
+        part = 911
+        offset = 10000
+        reqs_ds = [Deferred(), Deferred()]
+        mockclient = Mock()
+        mockclient.send_offset_request.side_effect = reqs_ds
+        consumer = Consumer(mockclient, topic, part, Mock())
+        d = consumer.start(offset)
+
+        fetch_request = FetchRequest(topic=topic, partition=part, offset=offset,
+                                     max_bytes=FETCH_BUFFER_SIZE_BYTES)
+        consumer.client.send_fetch_request.assert_called_once_with([fetch_request], max_wait_time=100, min_bytes=65536)
+
+        f = Failure(OffsetOutOfRangeError())
+        reqs_ds[0].errback(f)
+
+        # fetch offset
```

Comment:
@rthille  I need some help regarding mocking the trigger of OffsetOutOfRangeError()  to check the fire of or calling on `send_offset_request` with `fetch_request`.